sdk/storage/azure-storage-file/apireview/azure/storage/file/aio/share_client_async.pyi

```diff
@@ -0,0 +1,35 @@
+# Stubs for azure.storage.file.aio.share_client_async (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from .._deserialize import deserialize_permission, deserialize_share_properties
+from .._generated.aio import AzureFileStorage
+from .._generated.models import DeleteSnapshotsOptionType, SignedIdentifier, StorageErrorException
+from .._generated.version import VERSION
+from .._shared.base_client_async import AsyncStorageAccountHostsMixin
+from .._shared.policies_async import ExponentialRetry
+from .._shared.request_handlers import add_metadata_headers, serialize_iso
+from .._shared.response_handlers import process_storage_error, return_headers_and_deserialized, return_response_headers
+from ..models import AccessPolicy, ShareProperties
+from ..share_client import ShareClient as ShareClientBase
+from .directory_client_async import DirectoryClient
+from .file_client_async import FileClient
+from typing import Any, Dict, Optional, Union
+
+class ShareClient(AsyncStorageAccountHostsMixin, ShareClientBase):
+    def __init__(self, share_url: str, share: Optional[Union[str, ShareProperties]]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., loop: Any=..., **kwargs: Any) -> None: ...
+    def get_directory_client(self, directory_path: Optional[str]=...) -> DirectoryClient: ...
```

Comment:
Not sure what happen if the path is optional?

Reply:
Make path required? And accept None to make it backward compatible?

Reply:
Note: Also look at pathlib behaviour